lib/bridge/MatrixHandler.js

```diff
@@ -19,6 +19,7 @@ const MSG_PMS_DISABLED_FEDERATION = "[Bridge] Sorry, PMs are disabled on " +
 
 const KICK_RETRY_DELAY_MS = 15000;
 const KICK_DELAY_JITTER = 30000;
+const EVENT_CACHE_SIZE = 4096;
```

Comment:
I doubt this is enough for Freenode. Matrix.org is handling ~25Hz of messages, so we'll round up to 30 for math+future capabilities. At 30Hz, this cache is overturned every 3 minutes. Given most replies I've seen (note: not actually tested) happen 5-10+ minutes after the original message, your cache is too small.

Having a large cache means you're more likely to run out of memory though. Combined with the fact that the cache is not self-expiring, you'll never reclaim that memory during low volume periods.

Reply:
This is assuming a lot of things.
> Matrix.org is handling ~25Hz of messages.

You're assuming the bridge can *see* 25Hz of events. Even freenode isn't in that many rooms and is excluding things like PM traffic and the ~90% of rooms that aren't in a bridge. Let's not also forget that some of that traffic is IRC, which won't hit the cache. I seriously doubt the bridge is seeing an *average* of 30Hz M->I traffic a second. 

>likely to run out of memory though

We could up this to 16384 and be laughing to be honest, it's not a lot of memory.

>cache is not self-expiring

It's really not something I want to waste CPU cycles on.

Reply:
Sorry, you're right - I was thinking of a similar nightmare from a different project and started having flashbacks. Would be nice to have the cache size configurable.